### DISCUSSION TITLE
Bump min. required JDK version to recent

### DISCUSSION THREAD

**Discussion by MarcinOrlowski** [logisim-evolution/logisim-evolution#951] [Dev talk]
I know we [just bumped JDK from 11 to 14](https://github.com/logisim-evolution/logisim-evolution/issues/789), but right now I am looking at v15 because of [text blocks](https://docs.oracle.com/en/java/javase/13/text_blocks/index.html) that is officially part of the language [starting v15](https://www.oracle.com/java/technologies/javase/15-relnote-issues.html). Using text blocks would significantly improve HDL template maintenance. 

In #952 ported some HDL in generators to use text blocks to show the difference. The additional nice thing about text blocks is that you do not need to escape quotes any more :) For example: [RandomHDLGeneratorFactory.java](https://github.com/logisim-evolution/logisim-evolution/pull/952/files#diff-2a676bb474467a3425659da8b182017e1809c86a724362dde49594e6f2acd74b)

Thougts?

@BFH-ktt1 @maehne @davidhutchens @zdimension

**Comment by zdimension** [comment]
Personally I'm always on favour of using the most up-to-date language features. It may be interesting though to consider the market share of different Java versions; is Java 15 widely enough used to bump the requirement?

**Reply by davidhutchens** [reply]
I should note that the nightly is using Java 14 on macOS. I don't know if you have determined if a later version is available. You should check before pushing the Java version.

Otherwise, I have no strong feeling about it.

**Reply by MarcinOrlowski** [reply]
The latest stable version is 16 which is available for download for all supported platforms: https://www.oracle.com/java/technologies/javase-jdk16-downloads.html

**Reply by BFH-ktt1** [reply]
Basically I do not have a problem with it, but seen that the semester starts in a few weeks I would propose:
1) Let's try to get the current development branch stable (hence do the required testing).
2) Bump out a new version that goes to master.
3) Continue our current efforts to clean-up the code.

I do not know what you all think of this proposal?

**Reply by MarcinOrlowski** [reply]
I think that the JDK change can be done now. I am now even thinking about going for JDK16 instead. It's not that its features are needed right away, but JDK15 is already EoL. JDK16 is available for download for any platform needed + it adds some plaform related fixes here and there. The JDK change do not require any code refactoring.

**Reply by MarcinOrlowski** [reply]
#957 bumps JDK to v16.

**Reply by MarcinOrlowski** [reply]
Ping. Let's resume and discuss JDKv16 being our minimum

**Comment by MarcinOrlowski** [comment]
To get you more convinced, I ported almost everything that is worth porting to text blocks as part of #952. PTAL
